EIPS/eip-7779.md

```diff
@@ -0,0 +1,106 @@
+---
+eip: 7779
+title: Controlled Gas Limit Increase
+description: Gradual Gas Limit Increase with Scheduled Deactivation Block
+author: Giulio Rebuffo (@Giulio2002)
+discussions-to: https://ethresear.ch/t/eip-7779-discussion-thread/20574
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-10-06
+---
+
+This proposal introduces a mechanism to gradually increase the block gas limit over two years, doubling from 30 million to 60 million gas, with an automatic deactivation at a specified block height. The purpose is to provide a controlled and predictable growth of block capacity to meet increasing network demand while ensuring network stability.
+
+## **Abstract**
+
+The EIP proposes a system to increase the block gas limit by a fixed amount at regular intervals over a two-year period. The block gas limit will increase incrementally from 30 million gas to 60 million gas, with a safeguard "deactivation block" that halts the increase after two years. This system is designed to accommodate network growth without imposing abrupt changes, giving developers, node operators, and users time to adapt. The incremental increase is fully automated, ensuring transparent and predictable block size growth.
+
+## **Motivation**
+
+### **Predictable Gas Limit Growth**
+
+- **Current Issue:**
+  - The Ethereum network faces increasing demand, but changes to the gas limit are often manually adjusted by miners or validators based on their preferences, which may cause unpredictable block sizes and network performance issues.
+
+- **Need for Change:**
+  - A systematic and predictable increase of the gas limit will help scale the network while giving the ecosystem time to adjust to larger block sizes, without needing to rely on ad hoc decisions by network participants.
+
+### **Gradual Increase with Deactivation Safeguard**
+
+- **Controlled Growth:**
+  - Instead of sudden or unpredictable changes, this EIP proposes incremental gas limit increases over two years, which will gradually double the block capacity.
+  
+- **Automatic Deactivation:**
+  - A safeguard deactivation block will halt the increase after two years, ensuring that the gas limit does not grow indefinitely without further consensus from the community.
+
+## **Specification**
+
+### **Incremental Gas Limit Increase**
+
+1. **Initial Block Gas Limit:**
+   - At the start of the proposal, the block gas limit is set to 30 million gas.
+
+2. **Increase Rate:**
+   - The block gas limit will increase by 0.00475% per block (approximately 1.5% per day), such that after approximately 730 days (two years), the gas limit reaches 60 million.
```

Comment:
For non-integer outputs of the exponential formula, we should round the result down to the nearest integer - right?

Reply:
Yes